### ISSUE TITLE
Remote Store but Local Search / Sort

### ISSUE THREAD

**Issue by naorrosenberg** [AllenFang/react-bootstrap-table#990]
I have a remote store (using MobX) and inject the data to the table component as props. I set the `remote` flag in the table accordingly.
Obviously I need to write my own `onAddRow`, `onDeleteRow` and `onCellEdit` to update the remote store, but I'd like to avoid rewriting the search and sort code.
Is there a way to use the default search / sort functionality with remote store?

**Comment by AllenFang** [comment]
@naorrosenberg, actually, it cant be done in current version, because inside the `react-bootstrap-table`, I always force a `return` if `remote` enabled for these operation. For example, [this](https://github.com/AllenFang/react-bootstrap-table/blob/master/src/BootstrapTable.js#L396) is a case of sorting.

However, I think your consideration is possible happen for the other user too, so I will gonna to support this feature and my solution will be like following:

```js

remote = (remoteFunction) => {
    // remoteFunction is a object, default is all false, 
    //  { insertRow: false, dropRow: false, cellEdit: false, sort: false, pagination: false, search: false, filter: false    }
    //
    remoteFunction.insertRow = true;
    remoteFunction.dropRow = true;
    remoteFunction.cellEdit = true;
    return remoteFunction;
}

render() {
    return {
        <BootstrapTable data={data} remote={this.getRemote}
    };
}
```

How about above example? is that much intuitive?

**Comment by naorrosenberg** [comment]
Thanks @AllenFang for such a prompt reply!

Yes - this solution would be great, thanks.
Do you expect this to enter the next beta version?

**Comment by AllenFang** [comment]
I can support this feature on next version(next week)

**Comment by AllenFang** [comment]
Fixed on `v2.11.1`, check https://github.com/AllenFang/react-bootstrap-table/blob/master/examples/js/remote/remote-store-alternative.js and https://github.com/AllenFang/react-bootstrap-table/blob/master/examples/js/remote/remote-alternative.js for how to use it, let me know if you got any trouble, thanks 👍

**Comment by tommiii** [comment]
I'm not able to set `defaultSearch` in options when remote is enable. I get 'Maximum update depth exceeded' error. Is there a solution in "react-bootstrap-table": "4.3.1" ?
